aiida/backends/managers/access.py

```diff
@@ -0,0 +1,259 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Module for the manager that tracks the access and locking of the profile."""
+import contextlib
+import json
+import os
+import time
+
+import psutil
+
+from aiida.common import exceptions
+
+
+class ProcessData():
+    """Class to store and manage the data of processes"""
+
+    def __init__(self, pid=None, filepath=None):
+        """Initialization process for the ProcessData object.
+
+        :param pid:
+            pid of a process to be loaded from the ones running in the environment (if it is not
+            provided it will just load the data of the process that is currently running).
+
+        :param filepath:
+            filepath of the process to be loaded from a file (if it is not provided it will just
+            load the data of the process that is currently running).
+        """
+
+        if filepath is not None:
+            self._read_from_file(filepath)
+            return
+
+        if pid is None:
+            pid = os.getpid()
+
+        self._pid = pid
+        process_obj = psutil.Process(pid)
+        process_cmd = process_obj.cmdline()
+        process_ctm = time.localtime(process_obj.create_time())
```

Comment:
This line is useless as it's replaced by the next one? Also, I would call it with a more descriptive name (process_ctime)